eZ/Publish/Core/FieldType/Image/IO/Legacy.php

```diff
@@ -149,7 +150,13 @@ public function loadBinaryFile($binaryFileId)
     public function loadBinaryFileByUri($binaryFileUri)
     {
         try {
-            return $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+            $image = $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+
+            if ($image instanceof MissingBinaryFile) {
+                $image = $this->draftIOService->loadBinaryFileByUri($binaryFileUri);
+            }
+
+            return $image;
         } catch (InvalidArgumentException $prefixException) {
             // InvalidArgumentException means that the prefix didn't match, NotFound can pass through
             try {
```

Comment:
yep. @glye you were right. 
`IOService` in its implementation of `loadBinaryFileByUrl` calls [loadBinaryFile](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/IO/IOService.php#L174) which then calls [checkBinaryFileId](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/IO/IOService.php#L157) which [throws](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/IO/IOService.php#L309) `InvalidBinaryFileIdException` and it [extends](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/IO/Exception/InvalidBinaryFileIdException.php#L13) `InvalidArgumentValue` which extends `InvalidArgumentException`. 

So in the end `loadBinaryFileByUrl` *can* throw `InvalidArgumentException` hence we cannot change docblock of `IOServiceInterface` and we cannot simplify (remove `catch` part) of `Legacy`